pkg/types/ovirt/defaults/machinepools.go

```diff
@@ -0,0 +1,42 @@
+package defaults
+
+import (
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/ovirt"
+)
+
+const (
+	//DAffinyGroupBase is a holder which will be replaced with the cluster name when creating the affinity group
+	DAffinyGroupBase                 = "CLUSTERNAME"
+	dControlPlaneAffinyGroupPostfix  = "-controlplane"
+	dControlPlaneAffinyGroupPriority = 5
+	dComputeAffinyGroupPostfix       = "-cluster"
+	dComputeAffinyGroupPriority      = 3
+)
+
+func setMachinePool(p *types.MachinePool) {
+	if p.Platform.Ovirt == nil {
+		p.Platform.Ovirt = &ovirt.MachinePool{}
+	}
+}
+
+// SetControlPlaneDefaults sets the defaults for the ControlPlane Machines.
+func SetControlPlaneDefaults(p *types.MachinePool) {
+	setMachinePool(p)
+	if p.Platform.Ovirt.AffinityGroups == nil {
```

Comment:
```suggestion
	if len(p.Platform.Ovirt.AffinityGroups) == 0 {
```

Reply:
I'm ok with having 0 size, in that case no affinity groups will be created, that is the customization that will disable the affinity groups

Reply:
A roundtrip marshaling and unmarshaling of the install config is going to lose the difference between a missing `AffinityGroups` field and an `AffinityGroups` field set to an empty list.

Take for example an install-config.yaml with the following `controlPlane` section.
```
controlPlane:
  platform:
    ovirt:
      affinityGroups: []
```

Now, run `openshift-install create install-config`. The `controlPlane` section will become the following.
```
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    ovirt: {}
  replicas: 3
```

Now, run `openshift-install create install-config` again. The `controlPlane` section will become the following.
```
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    ovirt:
      affinityGroups:
      - name: CLUSTERNAME-controlplane
        priority: 5
      - name: CLUSTERNAME-cluster
        priority: 3
  replicas: 3
```

Reply:
Ok I see I guess we should handle that case, and when a user enters an empty array we shouldn't change the yaml and keep the empty array.
Great catch